### PR TITLE
Improve Android TV home screen recommendations

### PR DESCRIPTION
**Pull request by OctoNezd**
**Changes**

PR introduces 3 new home screen channels:
- Latest media - new default channel
- Movies
- New episodes
image

**Issues**

#2513

### CODE REVIEW

**Comment by OctoNezd** [comment]
Also there probably should be option to make channel empty, considering that it cant be disabled at all on GoogleTV - I will implement it

File: app/src/main/java/org/jellyfin/androidtv/integration/LeanbackChannelWorker.kt [hunk 1]
```diff
@@ -216,73 +282,143 @@
...
+					isPlayed = false
+				).content.orEmpty()
+			}
+			// Concat
+			Triple(latestEpisodes.await(), latestMovies.await(), latestMedia.await())
+		}
 
-			val seasonString = item.parentIndexNumber?.toString().orEmpty()
+	@SuppressLint("RestrictedApi")
+	private fun createPreviewProgram(
```
[the remark this answers is not in the document]
Comment (OctoNezd) [reply app/src/main/java/org/jellyfin/androidtv/integration/LeanbackChannelWorker.kt:353]: I don't see how this method can be simplified or expanded into multiple ones

**Review by nielsvanvelzen — CHANGES_REQUESTED** [review]
Thanks for the pull request. There are two major issues that need to be changed before I can start a code review:

- Remove the "Display backdrop" option
  The backdrops are not meant to be used as thumbnails and they are not always in 16:9 aspect ratio. The Android TV launcher warps the image to fit the 16/9 ratio when the backdrop is a different size which looks awful.

- Remove "leanbackPreferredDefaultChannel" option
  There can be only one default channel and it's set when the app first starts so changing it is just not possible. Your approach to change the reported values based on the preference is a hack that I don't want to support. This is a bit unfortunate for Google TV users but there's nothing we can do about it.

**Comment by OctoNezd** [comment]
1) Maybe change it to "Prefer thumbnails" instead then

2) Maybe change the default channel to something more useful than just media collections, e.g. just latest media?

**Comment by OctoNezd** [comment]
> * The backdrops are not meant to be used as thumbnails and they are not always in 16:9 aspect ratio. The Android TV launcher warps the image to fit the 16/9 ratio when the backdrop is a different size which looks awful.

We can requested a cropped backdrop instead through fillHeight and fillWidth? 

In addition to:

> The backdrops are not meant to be used as thumbnails 

Jellyfin web ui calls backdrops "thumbs"
image

**Comment by nielsvanvelzen** [comment]
> We can requested a cropped backdrop instead through fillHeight and fillWidth?

We try to use the least amount of parameters possible so we use our local image cache, so not a huge fan of this approach.

**Comment by OctoNezd** [comment]
The lack of prefer backdrops option results in ugly letterboxing on gtv devices, and having it results in ugly stretching on atv then. This is a bit problematic situation, both outcomes end up in situation where one part of userbase gets screwed :/

**Comment by OctoNezd** [comment]
Weird - the channel shows up for me on google tv emulator created with ADK. No next up/continue watching though. Did you add Google Account? Check if you are running apps-only mode under your google account settings. Also, apparently GTV uses channels/recommendations only in specific regions, where "the full google tv experience" is not available (cant find source for this statement, it was in some atv documentation by google).

**Comment by OctoNezd** [comment]
image
Here is a screenshot of gtv recommendations when using posters

**Comment by nielsvanvelzen** [comment]
>  Also, apparently GTV uses channels/recommendations only in specific regions, where "the full google tv experience" is not available (cant find source for this statement, it was in some atv documentation by google).

I think this might be the issue then because I'm getting the full advertisement experience (Netherlands).

**Comment by OctoNezd** [comment]
> > Also, apparently GTV uses channels/recommendations only in specific regions, where "the full google tv experience" is not available (cant find source for this statement, it was in some atv documentation by google).
> 
> I think this might be the issue then because I'm getting the full advertisement experience (Netherlands).

Yep - https://www.youtube.com/watch?v=O3BhUSEu8gg


8:51
You may have noticed home screen channels weren’t on the earlier Google TV screenshot.
8:56
For Google TV, home screen channels are only shown in markets
9:00
where we don't have enough information about content
9:02
to power the full Google TV experience.
9:05
That means they're not going to show up in markets like the US.

**Comment by OctoNezd** [comment]
So - both options need to go then, I assume?

**Comment by nielsvanvelzen** [comment]
Yup

**Comment by OctoNezd** [comment]
All new options were removed

**Review by nielsvanvelzen — APPROVED** [review]
Thanks! Everything is good now. I've made some final small tweaks that you can see in the last commit.

**Pull request merged**